examples/hif2a/fit_to_multiple_rbfes.py

```diff
@@ -292,22 +300,11 @@ def _blew_up(results: List[SimulationResult]):
         # TODO: adjust this threshold a bit, move reliability calculations into fe/estimator.py or fe/model.py
         return np.isnan(du_dls).any() or (du_dls.std(1).max() > 1000)
 
-
-    def _stringify_key(key: Tuple[int, str]):
-        return f'{key[0]}_{key[1]}'
-
-
-    flat_theta_traj = []
-    flat_grad_traj = []
-    loss_traj = []
-
-    results_traj = dict() # indexed by (step, stage) tuples # TODO: proper type hint
-
+    results_this_step = dict() # {stage : result} pairs # TODO: proper type hint
 
     def save_in_memory_callback(results, stage):
-        # TODO: replace with saving this to disk rather than accumulating in memory
-        global results_traj
-        results_traj[(step, stage)] = results
+        global results_this_step
```

Comment:
Random Q: do you actually need to declare the global here? Doesn't this just form a closure around the variable anyways?

Reply:
Probably not, just emphasizing the intent to have this function modify in-place a variable in main scope, even though it's going to be called a [couple](https://github.com/proteneer/timemachine/blob/0ce5b98dcadf5eb8ee54b1d069d22f6bf8792e7d/fe/model.py#L131) [layers](https://github.com/proteneer/timemachine/blob/0ce5b98dcadf5eb8ee54b1d069d22f6bf8792e7d/fe/estimator.py#L112) deep... (This callback workaround is temporary, until we settle on a Jax-idiomatic return type for deltaG() that more fully captures what we want to access from a free energy simulation.)